msi/createmsi.py

```diff
@@ -0,0 +1,139 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys, os, subprocess, shutil, uuid
+import platform
+
+sys.path.append(os.getcwd())
+from mesonbuild import coredata
+
+xml_templ = '''<?xml version='1.0' encoding='windows-1252'?>
+<Wix xmlns='http://schemas.microsoft.com/wix/2006/wi'>
+  <Product Name='Meson Build System' Manufacturer='The Meson Development Team'
+           Id='%s' 
+           UpgradeCode='%s'
+           Language='1033' Codepage='1252' Version='%s'>
+    <Package Id='*' Keywords='Installer' Description="Meson %s installer"
+             Comments='Meson is a high performance build system' Manufacturer='Meson development team'
+             InstallerVersion='100' Languages='1033' Compressed='yes' SummaryCodepage='1252' />
+
+    <Media Id="1" Cabinet="meson.cab" EmbedCab="yes" />
+
+    <Directory Id='TARGETDIR' Name='SourceDir'>
+      <Directory Id="ProgramFilesFolder">
+        <Directory Id="INSTALLDIR" Name="Meson">
+'''
+
+xml_footer_templ = '''
+        </Directory>
+      </Directory>
+    </Directory>
+
+    <Feature Id="DefaultFeature" Level="1">
+%s
+    </Feature>
+
+  <Property Id="WIXUI_INSTALLDIR" Value="INSTALLDIR" />
+  <UIRef Id="WixUI_InstallDir" />
+
+  </Product>
+</Wix>
+'''
+
+file_templ = '''<File Id='%s' Name='%s' DiskId='1' Source='%s'  />
+'''
+
+comp_ref_templ = '''<ComponentRef Id="%s" />
+'''
+
+def gen_guid():
+    return str(uuid.uuid4()).upper()
+
+class Node:
+    def __init__(self, dirs, files):
+        assert(isinstance(dirs, list))
+        assert(isinstance(files, list))
+        self.dirs = dirs
+        self.files = files
+
+class PackageGenerator:
+
+    def __init__(self):
+        self.version = coredata.version
+        self.guid = 'DF5B3ECA-4A31-43E3-8CE4-97FC8A97212E'
+        self.update_guid = '141527EE-E28A-4D14-97A4-92E6075D28B2'
+        self.main_xml = 'Meson.wxs'
+        self.main_o = 'Meson.wixobj'
+        self.bytesize = '32' if '32' in platform.architecture()[0] else '64'
+        self.final_output = 'meson-%s-%s.msi' % (self.version, self.bytesize)
+        self.staging_dir = 'dist'
+        if not os.path.exists(os.path.join(self.staging_dir, 'meson.exe')):
+            sys.exit('Meson exe missing from staging dir.')
+        if not os.path.exists(os.path.join(self.staging_dir, 'ninja.exe')):
+            sys.exit('Ninja exe missing from staging dir.')
+                              
+
+    def generate_files(self):
+        assert(os.path.isdir(self.staging_dir))
+        #cumulative_xml = ''
```

Comment:
__[flake8]__
* [E265] block comment should start with '# '

<!-- Comment by [SideCI](https://sideci.com) -->